tests/TestCase/ORM/EntityTest.php

```diff
@@ -46,6 +46,31 @@ public function testSetOneParamNoSetters()
         $this->assertSame(1, $entity->id);
         $this->assertEquals(1, $entity->getOriginal('id'));
         $this->assertEquals('bar', $entity->getOriginal('foo'));
+
+        $entity->set('level1.level2.level3', 'three');
+        $this->assertSame('three', $entity->level1->level2->level3);
+        $this->assertEquals('three', $entity->getOriginal('level1.level2.level3'));
+        $entity->set('level1.level2.level3', '333');
+        $this->assertEquals('three', $entity->getOriginal('level1.level2.level3'));
```

Comment:
Totally agree, but the question is _how_ to stop when we run out of Entities (but could theoretically descend into an array that lacks the ability to track dirty state.) 

Is that an Exception or does that short-circuit at the farthest Entity down the chain and return true?